clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -127,97 +37,66 @@
-    DESCRIBE_CONFIGS(32, "DescribeConfigs", DescribeConfigsRequest.schemaVersions(),
-            DescribeConfigsResponse.schemaVersions()),
-    ALTER_CONFIGS(33, "AlterConfigs", AlterConfigsRequest.schemaVersions(),
-            AlterConfigsResponse.schemaVersions()),
-    ALTER_REPLICA_LOG_DIRS(34, "AlterReplicaLogDirs", AlterReplicaLogDirsRequest.schemaVersions(),
-            AlterReplicaLogDirsResponse.schemaVersions()),
-    DESCRIBE_LOG_DIRS(35, "DescribeLogDirs", DescribeLogDirsRequest.schemaVersions(),
-            DescribeLogDirsResponse.schemaVersions()),
-    SASL_AUTHENTICATE(36, "SaslAuthenticate", SaslAuthenticateRequestData.SCHEMAS,
-            SaslAuthenticateResponseData.SCHEMAS),
-    CREATE_PARTITIONS(37, "CreatePartitions", CreatePartitionsRequest.schemaVersions(),
-            CreatePartitionsResponse.schemaVersions()),
-    CREATE_DELEGATION_TOKEN(38, "CreateDelegationToken", CreateDelegationTokenRequestData.SCHEMAS, CreateDelegationTokenResponseData.SCHEMAS),
-    RENEW_DELEGATION_TOKEN(39, "RenewDelegationToken", RenewDelegationTokenRequestData.SCHEMAS, RenewDelegationTokenResponseData.SCHEMAS),
-    EXPIRE_DELEGATION_TOKEN(40, "ExpireDelegationToken", ExpireDelegationTokenRequestData.SCHEMAS, ExpireDelegationTokenResponseData.SCHEMAS),
-    DESCRIBE_DELEGATION_TOKEN(41, "DescribeDelegationToken", DescribeDelegationTokenRequestData.SCHEMAS, DescribeDelegationTokenResponseData.SCHEMAS),
-    DELETE_GROUPS(42, "DeleteGroups", DeleteGroupsRequestData.SCHEMAS, DeleteGroupsResponseData.SCHEMAS),
-    ELECT_LEADERS(43, "ElectLeaders", ElectLeadersRequestData.SCHEMAS,
-            ElectLeadersResponseData.SCHEMAS),
-    INCREMENTAL_ALTER_CONFIGS(44, "IncrementalAlterConfigs", IncrementalAlterConfigsRequestData.SCHEMAS,
-                              IncrementalAlterConfigsResponseData.SCHEMAS),
-    ALTER_PARTITION_REASSIGNMENTS(45, "AlterPartitionReassignments", AlterPartitionReassignmentsRequestData.SCHEMAS,
-                                  AlterPartitionReassignmentsResponseData.SCHEMAS),
-    LIST_PARTITION_REASSIGNMENTS(46, "ListPartitionReassignments", ListPartitionReassignmentsRequestData.SCHEMAS,
-                                 ListPartitionReassignmentsResponseData.SCHEMAS);
-
-    private static final ApiKeys[] ID_TO_TYPE;
-    private static final int MIN_API_KEY = 0;
-    public static final int MAX_API_KEY;
+    PRODUCE(0),
+    FETCH(1),
+    LIST_OFFSETS(2),
+    METADATA(3),
+    LEADER_AND_ISR(4),
+    STOP_REPLICA(5),
+    UPDATE_METADATA(6),
+    CONTROLLED_SHUTDOWN(7),
+    OFFSET_COMMIT(8),
+    OFFSET_FETCH(9),
+    FIND_COORDINATOR(10),
+    JOIN_GROUP(11),
+    HEARTBEAT(12),
+    LEAVE_GROUP(13),
+    SYNC_GROUP(14),
+    DESCRIBE_GROUPS(15),
+    LIST_GROUPS(16),
+    SASL_HANDSHAKE(17),
+    API_VERSIONS(18),
+    CREATE_TOPICS(19),
+    DELETE_TOPICS(20),
+    DELETE_RECORDS(21),
+    INIT_PRODUCER_ID(22),
+    OFFSET_FOR_LEADER_EPOCH(23),
+    ADD_PARTITIONS_TO_TXN(24),
```

Comment:
I think this one and `ADD_OFFSETS_TO_TXN` should have magic v2 set has well.

Reply:
thanks.  sorry about the transcription error